lib/events/stream.go

```diff
@@ -393,12 +389,9 @@ func (s *ProtoStream) EmitAuditEvent(ctx context.Context, event apievents.AuditE
 func (s *ProtoStream) Complete(ctx context.Context) error {
 	s.complete()
 	select {
-	// wait for all in-flight uploads to complete and stream to be completed
-	case <-s.uploadsCtx.Done():
+	case <-s.uploadLoopDoneCh:
 		s.cancel()
 		return s.getCompleteResult()
-	case <-s.cancelCtx.Done():
```

Comment:
this is already handled everywhere this is necessary, handling it here will do the wrong thing as well - it will not allow the `receiveAndUpload` to complete before the streamer is closed.